src/api/core/warren/tests/test_indicators_mixins.py

```diff
@@ -0,0 +1,388 @@
+    """Test saving cached results to the database."""
+
+    class MyIndicator(CacheMixin, BaseIndicator):
+        """Dummy indicator."""
+
+        def get_lrs_query(self) -> LRSQuery:
+            return LRSQuery(query={"verb": "played"})
+
+        async def fetch_statements(self):
+            pass
+
+        async def compute(self):
+            pass
+
+    indicator = MyIndicator()
+    assert (
+        len(
+            db_session.exec(select(Cache).where(Cache.key == indicator.cache_key)).all()
+        )
+        == 0
+    )
+    await indicator.save(Cache(key=indicator.cache_key, value={"lol": [1, 2, 3]}))
+
+    saved = db_session.exec(select(Cache).where(Cache.key == indicator.cache_key)).one()
+
+    assert isinstance(saved, Cache)
+    assert saved.key == indicator.cache_key
+    assert saved.value == {"lol": [1, 2, 3]}
+    assert saved.since is None
+    assert saved.until is None
+    assert saved.created_at == datetime(2023, 10, 14, tzinfo=timezone.utc)
+
+
+@pytest.mark.anyio
+async def test_save_with_multiple_cache_instances(db_session):
+    """Test saving cached results to the database (multiple values case)."""
+
+    class MyIndicator(CacheMixin, BaseIndicator):
+        """Dummy indicator."""
+
+        def get_lrs_query(self) -> LRSQuery:
+            return LRSQuery(query={"verb": "played"})
+
+        async def fetch_statements(self):
+            pass
+
+        async def compute(self):
+            pass
+
+    indicator = MyIndicator()
+    assert (
+        len(
+            db_session.exec(select(Cache).where(Cache.key == indicator.cache_key)).all()
+        )
+        == 0
+    )
+    await indicator.save(
+        [
+            Cache(key=indicator.cache_key, value={"bar": [1, 2, 3]}),
+            Cache(key=indicator.cache_key, value={"bar": [4, 5, 6]}),
+        ]
+    )
+
+    saved = db_session.exec(
+        select(Cache).where(Cache.key == indicator.cache_key).order_by("created_at")
+    ).all()
+
+    assert isinstance(saved, list)
+    assert saved[0].key == indicator.cache_key
+    assert saved[0].value == {"bar": [1, 2, 3]}
+    assert saved[0].since is None
+    assert saved[0].until is None
+    assert saved[1].key == indicator.cache_key
+    assert saved[1].value == {"bar": [4, 5, 6]}
+    assert saved[1].since is None
+    assert saved[1].until is None
+
+
+@pytest.mark.anyio
+async def test_get_cache(db_session):
+    """Test getting cached results from the database."""
+
+    class MyIndicatorA(CacheMixin, BaseIndicator):
+        """Dummy indicator."""
+
+        def get_lrs_query(self) -> LRSQuery:
+            return LRSQuery(query={"verb": "played"})
+
+        async def fetch_statements(self):
+            pass
+
+        async def compute(self):
+            pass
+
+    class MyIndicatorB(CacheMixin, BaseIndicator):
+        """Dummy indicator."""
+
+        def get_lrs_query(self) -> LRSQuery:
+            return LRSQuery(query={"verb": "viewed"})
```

Comment:
Demonstrate that even if we have the same LRS query, the cache key differs:

```suggestion
            return LRSQuery(query={"verb": "played"})
```